airflow/models/taskinstance.py

```diff
@@ -2216,7 +2216,10 @@ def _check_and_change_state_before_execution(
 
         ti.state = TaskInstanceState.RUNNING
         ti.emit_state_change_metric(TaskInstanceState.RUNNING)
-        ti.external_executor_id = external_executor_id
+
+        if external_executor_id:
```

Comment:
Make sense. I believe this piece of code runs on the worker side. So, do we have any use cases  where we change `external_executor_id` value from worker side?

Reply:
That's a good question. A quick search appears we never do?